Shared/Samples/Style features with custom dictionary/StyleFeaturesWithCustomDictionaryView.swift

```diff
@@ -0,0 +1,137 @@
+            }
+            .pickerStyle(.segmented)
+            .padding()
+            .onChange(of: dictionaryStyleSelection) { _ in
+                // Update the feature layer with the correct dictionary renderer
+                // on selection change.
+                model.restaurantFeatureLayer.renderer = dictionaryStyleSelection == .web ? model.dictionaryRendererFromWebStyle : model.dictionaryRendererFromStyleFile
+            }
+        }
+    }
+}
+
+private extension StyleFeaturesWithCustomDictionaryView {
+    /// The view model for the sample.
+    class Model: ObservableObject {
+        /// A map with a topographic basemap centered on Esri in Redlands, CA.
+        let map: Map = {
+            let map = Map(basemapStyle: .arcGISTopographic)
+            map.initialViewpoint = Viewpoint(
+                latitude: 34.0543,
+                longitude: -117.1963,
+                scale: 1e4
+            )
+            return map
+        }()
+        
+        /// A feature layer with restaurants in Redlands, CA.
+        let restaurantFeatureLayer: FeatureLayer = {
+            let restaurantFeatureTable = ServiceFeatureTable(url: .redlandsRestaurants)
+            return FeatureLayer(featureTable: restaurantFeatureTable)
+        }()
+        
+        /// A dictionary renderer created from a custom symbol style dictionary file (.stylx) on local disk.
+        let dictionaryRendererFromStyleFile: DictionaryRenderer = {
+            // The URL to the symbol style dictionary file.
+            let restaurantStyleURL = URL.restaurants
+            
+            // Create the dictionary symbol style from the style file URL.
+            let restaurantStyle = DictionarySymbolStyle(url: restaurantStyleURL)
+            
+            // Create the dictionary renderer from the dictionary symbol style.
+            return DictionaryRenderer(dictionarySymbolStyle: restaurantStyle)
+        }()
+        
+        /// A dictionary renderer created from a custom symbol style hosted on ArcGIS Online.
+        let dictionaryRendererFromWebStyle: DictionaryRenderer = {
+            // The create a portal item with the id of the online web style.
+            let restaurantPortalItem = PortalItem(
+                portal: .arcGISOnline(connection: .anonymous),
+                id: .restaurantWebStyle
+            )
+            
+            // Create a dictionary symbol style from the web style portal item.
+            let restaurantStyle = DictionarySymbolStyle(portalItem: restaurantPortalItem)
+            
+            // Create a dictionary renderer from the dictionary symbol style.
+            // Map the input fields in the feature layer to the dictionary symbol
+            // style's expected fields for symbols.
+            return DictionaryRenderer(
+                dictionarySymbolStyle: restaurantStyle,
+                symbologyFieldOverrides: ["healthgrade": "Inspection"],
+                textFieldOverrides: [:]
```

Comment:
This parameter has a default value:
```suggestion
                symbologyFieldOverrides: ["healthgrade": "Inspection"]
```